fftest/vars.go

```diff
@@ -54,28 +54,24 @@ type Vars struct {
 // Compare one set of vars with another
 // and return an error on any difference.
 func Compare(want, have *Vars) error {
-	if want.WantParseErrorIs != nil || want.WantParseErrorString != "" {
-		if want.WantParseErrorIs != nil && have.ParseError == nil {
-			return fmt.Errorf("want error (%v), have none", want.WantParseErrorIs)
-		}
-
-		if want.WantParseErrorString != "" && have.ParseError == nil {
-			return fmt.Errorf("want error (%q), have none", want.WantParseErrorString)
-		}
+	if want.WantParseErrorIs != nil && have.ParseError == nil {
+		return fmt.Errorf("want error (%v), have none", want.WantParseErrorIs)
+	}
```

Comment:
You can't change the Compare test helper in this way. If Parse returns an error, the result of the Parse against the flagset is undefined behavior, so you can't continue to test the other conditions. 

This is what's triggering the bug in ffyaml: it decodes the file into a map, and then ranges over the map to process each value, but ranging over a map in Go yields keys in random order, so sometimes it gets the `i` value first, sometimes the `s`.

Reply:
Okay, let me try harder.

Reply:
I don't know what I was thinking. Just reverting my change to that file, and everything works as expected. Thanks!